File: backend/src/main/java/kr/co/kfs/asset/oms/domain/sys/service/interfaces/Sys02UserService.java
package kr.co.kfs.asset.oms.domain.sys.service.interfaces;

import kr.co.kfs.asset.oms.domain.sys.dto.Sys02UserDto;
import java.util.List;

public interface Sys02UserService {
    Sys02UserDto getById(Long userId);
    Sys02UserDto getByLoginId(String loginId);
    List<Sys02UserDto> getByCompanyId(Long companyId);
    int insert(Sys02UserDto dto);
    int update(Sys02UserDto dto);
    int delete(Long userId);
}

File: frontend/src/services/sys02UserService.ts
import api from './api'

export interface UserDto {
  userId?: number
  companyId: number
  loginId: string
  korNm: string
  passwd?: string
  email?: string
  adminYn?: string
  note?: string
  companyName?: string
}

export const sys02 = {
  getByCompany: (companyId: number) =>
    api.get<UserDto[]>('/api/v1/sys/users', { params: { companyId } }).then(r => r.data),
  getById: (id: number) => api.get<UserDto>(`/api/v1/sys/users/${id}`).then(r => r.data),
  insert: (dto: UserDto) => api.post<number>('/api/v1/sys/users', dto).then(r => r.data),
  update: (id: number, dto: UserDto) => api.put<number>(`/api/v1/sys/users/${id}`, dto).then(r => r.data),
  delete: (id: number) => api.delete<number>(`/api/v1/sys/users/${id}`).then(r => r.data),
}
